src/storage/src/hummock/sstable/block_iterator.rs

```diff
@@ -94,19 +99,27 @@ impl BlockIterator {
     }
 
     pub fn seek(&mut self, key: FullKey<&[u8]>) {
-        let full_key_encoded = key.encode();
-        self.seek_restart_point_by_key(&full_key_encoded);
-        self.next_until_key(&full_key_encoded);
+        if key.user_key.table_id.table_id() == self.block.table_id() {
```

Comment:
This is incorrect since we miss to handle the case when table id doesn't match. The semantic of seek is to seek to the first position in which the key >= the provided key.
- `if key.user_key.table_id.table_id() > self.block.table_id()`, we should `invalidate`
- `if key.user_key.table_id.table_id() < self.block.table_id()`, we should `seek_to_last`